### PR TITLE
2024 standard

### PR DESCRIPTION
**Pull request by aze12345**
Same as for 2024

### CODE REVIEW

File: custom/sensors/CustomCANCoder.java [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+// import com.ctre.phoenix.sensors.SensorTimeBase;
+// TODO: what is sensor time base why is it gone
 
 // import org.usfirst.frc4904.standard.LogKitten;
 import org.usfirst.frc4904.standard.Util;
 
-public class CustomCANCoder extends CANCoder implements CustomEncoder {
+public class CustomCANCoder extends CANcoder implements CustomEncoder {
```
Comment (Papayapaya) [inline custom/sensors/CustomCANCoder.java:11]: is CANcoder one word? the C is lowercase everywhere except the class name

File: custom/motorcontrollers/CANTalonFX.java [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+import com.ctre.phoenix6.signals.ForwardLimitValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import com.ctre.phoenix6.signals.ReverseLimitValue;
+import com.ctre.phoenix6.hardware.TalonFX;
 
-public class CANTalonFX extends WPI_TalonFX implements TalonMotorController {
-	protected static final NeutralMode DEFAULT_NEUTRAL_MODE 	= NeutralMode.Coast;
-	protected static final InvertType  DEFAULT_INVERT_TYPE  	= InvertType.FollowMaster;
+public class CANTalonFX extends TalonFX {
+	protected static final NeutralModeValue DEFAULT_NEUTRAL_MODE 	= NeutralModeValue.Coast;
```
Comment (annvoor) [inline custom/motorcontrollers/CANTalonFX.java:9]: should change that comment block, inverted is not a thing for the falcons here anymore i don't think

File: custom/sensors/CustomCANCoder.java (hunk 1 above)
Comment (annvoor) [reply custom/sensors/CustomCANCoder.java:11]: for future reference, i think it's CANcoder and not CANCoder bc the ctre imports are CANcoder, but i don't want to change tho whole class name and stuff so it stays

File: custom/sensors/CustomCANCoder.java [hunk 3]
```diff
@@ -82,7 +84,8 @@ public boolean getDirection() {
 
     @Override
     public boolean getDirectionSafely() throws InvalidSensorException {
-        return !reverseDirection == (getPosition() >= 0);
+        // getPosition() returns status signal, can't use >= on status signal
+        return !reverseDirection == (getPosition().getValue() >= 0);
```
Comment (annvoor) [inline custom/sensors/CustomCANCoder.java:88]: i think you can do getPosition().getValue() and then change what the function is returning to smth else and it works? i don't remember but it works in a different file (idk which one)

**Pull request merged**